src/pages/collective.js

```diff
@@ -38,14 +39,16 @@ class CollectivePage extends React.Component {
     const { LoggedInUser } = this.state;
 
     if (data.loading) return (<Loading />);
+    if (!data.Collective) {
+      window.location.replace(`/search?q=${get(data, 'variables.slug')}`);
```

Comment:
I think that would be better in `componentDidMount`.

Reply:
`componentDidMount` will be triggered after <Loading /> is rendered.

Reply:
Hmm, it may not work in `componentDidMount` if the data is still being fetched? 

Also, I would use `Router.push` (`import { Router } from '../server/pages'`) instead of `window.location.replace` so as to not trigger a full page reload.

Reply:
I don't think it's the good place to perform a client side redirect in the middle of the `render`. For the server side `render', it's just broken, for the client side `render`, we're interrupting in the middle which is weird.

If we put it in `componentDidMount`, it will be triggered after the the `render`, so there will be effectively be a flash with the `NotFound` component, but that's already the case if the server side rendering goes through.

For the data still being fetched, that might be effectively true in the case of a client only 404 (navigating from a normal page, navigating with a link to a 404 page). Not sure how to best handle the case.

Overall, from a UX perspective, I think I would stay away of redirecting (changing the URL), it's weird. We could display search results in the page without redirecting, or give an incentive to the user to move there (with a big link or a big button ).

Reply:
I've removed the automatic redirect and replaced it with a button for now. Need to have this 404 page fixed and out the door. It's been hanging there for too long.